tests/integration/test_properties.py

```diff
@@ -67,41 +68,49 @@ def test_archive_property_by_id(client, auth_headers, new_property, test_databas
     assert responseBadPropertyID.status_code == 400
 
-def test_archive_property_by_id(client, auth_headers, new_property, test_database):
-    response = client.get('/api/properties', headers=auth_headers["admin"])
-    propertiesInfo = response.get_json()['properties']
-    firstProperty = propertiesInfo[0]
-    firstPropertyId = firstProperty['id']
-    allPropertyIds = [p['id'] for p in propertiesInfo]
-
-    """The server responds with a 401 error if a non-admin tries to archive"""
-    responseNoAdmin = client.post(f'/api/properties/archive', json={})
-    assert responseNoAdmin == 401
-    assert responseNoAdmin.json == {'message': 'Missing authorization header'}
-
-    """The archive properties endpoint should return a 201 code when successful"""
-    responseSuccess = client.post(f'/api/properties/archive', json={'ids': [firstPropertyId]}, headers=auth_headers['admin'])
-    assert responseSuccess.status_code == 201
-
-    """The archive properties endpoint should return data for all properties (check count, spot-check some fields) when successful"""
-    responseSuccess = client.post(f'/api/properties/archive', json={'ids': [firstPropertyId]}, headers=auth_headers['admin'])
-    propertiesReturnedByEndpoint = json.loads(responseSuccess.data)['properties']
-    assert len(propertiesReturnedByEndpoint) == len(propertiesInfo)
-    assert all([ p['name'] and p['address'] for p in propertiesReturnedByEndpoint ])
-
-    """The (single) archived property has its 'archived' key set to True"""
-    responsePropertyByName = client.get(f'/api/properties/{firstProperty["name"]}', headers=auth_headers['admin'])
-    assert json.loads(responsePropertyByName.data)['archived']
-
-    """When archiving multiple properties, all properties have the 'archived' key set to True"""
-    responseSuccess = client.post(f'/api/properties/archive', json={'ids': allPropertyIds}, headers=auth_headers['admin'])
-    responseAllProperties = client.get(f'/api/properties', headers=auth_headers['admin'])
-    assert all([ prop['archived'] for prop in json.loads(responseAllProperties.data)['properties'] ])
-
-    """The server responds with a 400 error if request body contains a non-existent property id"""
-    responseBadPropertyID = client.post('/api/properties/archive', json={'ids': [99999]}, headers=auth_headers['admin'])
-    assert responseBadPropertyID.get_json() == {'message': 'Properties cannot be archived'}
-    assert responseBadPropertyID.status_code == 400
+@pytest.mark.usefixtures('client_class', 'empty_test_db')
+class TestPropertyArchivalMethods:
+    def test_archive_properties_non_admin(self, staff_header):
+        """The server responds with a 401 error if a non-admin tries to archive"""
+        responseNoAdmin = self.client.patch(f'/api/properties/archive', json={}, headers=staff_header)
+        assert responseNoAdmin == 401
+
+    def test_archive_properties(self, admin_header, create_property):
```

Comment:
FYI: The `admin_header` takes extra time to run, because it hits the auth endpoint each time it is used. So technically when the `auth_header` is used we are testing more than one endpoint. The `valid_header` was created to bypass that for speed and to ensure only the endpoint under test is being hit. 

The idea/goal is that the `valid_header` will be used for testing behavior. And then specific `admin_header`, `staff_header`, `pm_header` will be used for auth tests.